File: frontend/types/alert.ts
export type AlertType =
  | "low-stock"
  | "stockout-risk"
  | "reorder-required"
  | "overstock"
  | "slow-moving"
  | "forecast-anomaly";

export type AlertSeverity = "critical" | "warning" | "info";

export type AlertStatus = "new" | "acknowledged" | "resolved";

export interface Alert {
  id: string;
  type: AlertType;
  severity: AlertSeverity;
  sku: string;
  productName: string;
  category: string;
  supplier: string;
  message: string;
  recommendation: string;
  status: AlertStatus;
  createdAt: string;
  daysRemaining?: number;
}

export interface AlertFilters {
  type: string;
  severity: string;
  product: string;
  supplier: string;
  dateRange: string;
  search: string;
}

export interface AlertMetrics {
  totalActive: number;
  criticalCount: number;
  reorderRequired: number;
  overstockWarnings: number;
}

export interface AlertCategoryData {
  name: string;
  value: number;
  color: string;
}

export interface Recommendation {
  id: string;
  sku: string;
  productName: string;
  action: string;
  daysUntilAction: number;
  severity: AlertSeverity;
}

File: backend/app/schemas/alert.py
from __future__ import annotations

from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict

from app.models.alert import AlertSeverity, AlertStatus, AlertType


class AlertProductSummary(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    sku: str
    name: str
    category: Optional[str] = None
    supplier: Optional[str] = None


class AlertResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    product_id: str
    product: Optional[AlertProductSummary] = None
    type: AlertType
    severity: AlertSeverity
    title: str
    message: str
    recommendation: str
    status: AlertStatus
    created_at: datetime
    updated_at: datetime


class AlertListResponse(BaseModel):
    items: list[AlertResponse]
    total: int
    page: int
    page_size: int
    total_pages: int


class AlertGenerateResponse(BaseModel):
    generated: int
    skipped_duplicates: int
    message: str


class AlertUpdateRequest(BaseModel):
    status: AlertStatus


class AlertSummaryResponse(BaseModel):
    total_active: int
    critical: int
    warning: int
    info: int
    reorder_required: int
    low_stock: int
    stockout_risk: int
    overstock: int
    slow_moving: int
    forecast_anomaly: int